Spigot-Server-Patches/0399-implement-optional-per-player-mob-spawns.patch

```diff
@@ -643,36 +121,67 @@ index 79f854966..4100e367a 100644
          return this.bb;
      }
 diff --git a/src/main/java/net/minecraft/server/PlayerChunkMap.java b/src/main/java/net/minecraft/server/PlayerChunkMap.java
-index 85b21012e..46d205380 100644
+index 85b21012e..cf32abf12 100644
 --- a/src/main/java/net/minecraft/server/PlayerChunkMap.java
 +++ b/src/main/java/net/minecraft/server/PlayerChunkMap.java
-@@ -78,7 +78,8 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+@@ -78,7 +78,7 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
      private final PlayerMap playerMap;
      public final Int2ObjectMap<PlayerChunkMap.EntityTracker> trackedEntities;
      private final Queue<Runnable> z;
 -    private int viewDistance;
 +    int viewDistance; // Paper - private -> package private
-+    public final com.destroystokyo.paper.util.PlayerMobDistanceMap playerMobDistanceMap; // Paper
  
      // CraftBukkit start - recursion-safe executor for Chunk loadCallback() and unloadCallback()
      public final CallbackExecutor callbackExecutor = new CallbackExecutor();
-@@ -165,6 +166,24 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+@@ -108,6 +108,10 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+     // Paper start - distance maps
+     private final com.destroystokyo.paper.util.misc.PooledLinkedHashSets<EntityPlayer> pooledLinkedPlayerHashSets = new com.destroystokyo.paper.util.misc.PooledLinkedHashSets<>();
+ 
++    // Paper start - per player mob spawning
++    public final com.destroystokyo.paper.util.misc.PlayerAreaMap playerMobDistanceMap; // Paper // Note: This map is mostly managed in ChunkProviderServer#tickChunks
++    // Paper end - per player mob spawning
++
+     void addPlayerToDistanceMaps(EntityPlayer player) {
+         this.updateMaps(player);
+ 
+@@ -116,6 +120,9 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
+     }
+ 
+     void removePlayerFromDistanceMaps(EntityPlayer player) {
++        // Paper start - per player mob spawning
++        this.playerMobDistanceMap.remove(player);
```

Comment:
this.playerMobDistanceMap might be null here if per player mob spawn is disabled.

Reply:
correct